kroxylicious-filters/kroxylicious-sasl-inspection/src/main/java/io/kroxylicious/filters/sasl/inspection/SaslObserver.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filters.sasl.inspection;
+
+import org.apache.kafka.common.errors.AuthenticationException;
+
+/**
+ * A Sasl observer merely watches a SASL negotiation between client and server extracting the
```

Comment:
Let's mention that these things are: 
1. stateful with respect to a given channel
2. created via the factory class